qa/src/test/java/org/eclipse/kapua/service/device/steps/BrokerSteps.java

```diff
@@ -0,0 +1,280 @@
+     */
+    private static final long BROKER_STARTUP_WAIT_MILIS = 10_000;
+    /**
+     * Device birth topic.
+     */
+    private static final String MQTT_BIRTH = "$EDC/kapua-sys/rpione3/MQTT/BIRTH";
+    /**
+     * Device death topic.
+     */
+    private static final String MQTT_DC = "$EDC/kapua-sys/rpione3/MQTT/DC";
+    /**
+     * Access to device management service.
+     */
+    private DevicePackageManagementService devicePackageManagementService;
+    /**
+     * Device registry service for managing devices in DB.
+     */
+    private DeviceRegistryService deviceRegistryService;
+    /**
+     * Configuration service for kura devices.
+     */
+    private DeviceConfigurationManagementService deviceConfiguratiomManagementService;
+    /**
+     * Kura snapshot management service.
+     */
+    private DeviceSnapshotManagementService deviceSnapshotManagementService;
+    /**
+     * Kura bundle management service.
+     */
+    private DeviceBundleManagementService deviceBundleManagementService;
+    /**
+     * Service for issuing commnads on Kura device.
+     */
+    private DeviceCommandManagementService deviceCommandManagementService;
+    /**
+     * Factory for creating commands sent to Kura.
+     */
+    private DeviceCommandFactory deviceCommandFactory;
+    /**
+     * Embedded ActiveMQ broker.
+     */
+    private BrokerService broker;
+    /**
+     * Client simulating Kura device
+     */
+    private KuraDevice kuraDevice;
+
+    /**
+     * Scenario scoped step data.
+     */
+    private StepData stepData;
+
+    @Inject
+    public BrokerSteps(StepData stepData) {
+        this.stepData = stepData;
+    }
+
+    @Before
+    public void beforeScenario(Scenario scenario) throws Exception {
+        System.setProperty(DB_JDBC_CONNECTION_URL_RESOLVER.key(), "H2");
+
+        KapuaConfigurableServiceSchemaUtils.scriptSession(FULL_SCHEMA_PATH, DEFAULT_FILTER);
+
+        KapuaLocator locator = KapuaLocator.getInstance();
+        devicePackageManagementService = locator.getService(DevicePackageManagementService.class);
+        deviceRegistryService = locator.getService(DeviceRegistryService.class);
+        deviceConfiguratiomManagementService = locator.getService(DeviceConfigurationManagementService.class);
+        deviceSnapshotManagementService = locator.getService(DeviceSnapshotManagementService.class);
+        deviceBundleManagementService = locator.getService(DeviceBundleManagementService.class);
+        deviceCommandManagementService = locator.getService(DeviceCommandManagementService.class);
+        deviceCommandFactory = locator.getFactory(DeviceCommandFactory.class);
+
+        /*
+         * Initialize broker from MQ configuration file ad start it.
+         * After issuing start wait a moment to start completely.
+         */
+        broker = BrokerFactory.createBroker(new URI("xbean:activemq.xml"));
+        broker.start();
+        Thread.sleep(BROKER_STARTUP_WAIT_MILIS);
```

Comment:
I would suggest to use something like:

```
    private static void waitForPort(int port, Duration duration) throws InterruptedException {
        final Instant end = Instant.now().plus(duration);
        
        System.out.format ("Waiting for port %s ... %n", port);
        while ( true )  {
            try ( Socket socket = new Socket((String)null, port) ) {
                System.out.format ("Waiting for port %s ... done!%n", port);
                return;
            } catch (UnknownHostException e) {
                throw new RuntimeException ( e) ;
            } catch (IOException e) {
                // re-try
            }

            if ( Instant.now().isAfter(end) ) {
                throw new IllegalStateException(String.format ("Port %s did not open after %s seconds", duration.getSeconds()) );
            }
            
            Thread.sleep(1_000);
        }
    }
```

To wait for the broker to be ready, instead of a fixed timeout.